File: src/main/java/buttondevteam/discordplugin/commands/Command2DCSender.java
package buttondevteam.discordplugin.commands;

import buttondevteam.discordplugin.DPUtils;
import buttondevteam.lib.chat.Command2Sender;
import discord4j.core.object.entity.Message;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.val;

@RequiredArgsConstructor
public class Command2DCSender implements Command2Sender {
	private final @Getter
	Message message;

	@Override
	public void sendMessage(String message) {
		if (message.length() == 0) return;
		message = DPUtils.sanitizeString(message);
		message = Character.toLowerCase(message.charAt(0)) + message.substring(1);
		val msg = message;
		val author = this.message.getAuthorAsMember().block();
		if (author == null) return;
		this.message.getChannel().subscribe(ch -> ch.createMessage(author.getNicknameMention() + ", " + msg));
	}

	@Override
	public void sendMessage(String[] message) {
		sendMessage(String.join("\n", message));
	}
}

File: src/main/java/buttondevteam/discordplugin/commands/Command2DCSender.scala
package buttondevteam.discordplugin.commands

import buttondevteam.discordplugin.DPUtils
import buttondevteam.lib.chat.Command2Sender
import discord4j.core.`object`.entity.channel.MessageChannel
import discord4j.core.`object`.entity.{Message, User}

class Command2DCSender(val message: Message) extends Command2Sender {
    def getMessage: Message = this.message

    override def sendMessage(message: String): Unit = {
        if (message.isEmpty) return
        var msg = DPUtils.sanitizeString(message)
        msg = Character.toLowerCase(message.charAt(0)) + message.substring(1)
        this.message.getChannel.flatMap((ch: MessageChannel) => ch.createMessage(this.message.getAuthor.map((u: User) => DPUtils.nickMention(u.getId) + ", ").orElse("") + msg)).subscribe
    }

    override def sendMessage(message: Array[String]): Unit = sendMessage(String.join("\n", message: _*))

    override def getName: String = Option(message.getAuthor.orElse(null)).map(_.getUsername).getOrElse("Discord")
}